src/ui/public/vislib/lib/axis_title.js

```diff
@@ -44,30 +47,38 @@ export default function AxisTitleFactory(Private) {
    * @param title {String} Axis title
    * @returns {Function} Appends axis title to a D3 selection
    */
-  AxisTitle.prototype.draw = function (title) {
-    let self = this;
+  AxisTitle.prototype.draw = function () {
+    const self = this;
 
     return function (selection) {
       selection.each(function () {
-        let el = this;
-        let div = d3.select(el);
-        let width = $(el).width();
-        let height = $(el).height();
+        const el = this;
+        const div = d3.select(el);
+        const width = $(el).width();
+        const height = $(el).height();
 
         self.validateWidthandHeight(width, height);
 
-        div.append('svg')
+        const svg = div.append('svg')
         .attr('width', width)
-        .attr('height', height)
-        .append('text')
+        .attr('height', height);
+
+
+        const bbox = svg.append('text')
         .attr('transform', function () {
-          if (div.attr('class') === 'x-axis-title') {
+          if (self.axis.isHorizontal()) {
             return 'translate(' + width / 2 + ',11)';
           }
-          return 'translate(11,' + height / 2 + ')rotate(270)';
+          return 'translate(11,' + height / 2 + ') rotate(270)';
         })
         .attr('text-anchor', 'middle')
-        .text(title);
+        .text(self.title).node().getBBox();
```

Comment:
I didn't notice these other calls the first couple times I read through, maybe they should be on new lines